applications/GeoMechanicsApplication/tests/cpp_tests/test_time_loop_executor.cpp

```diff
@@ -358,4 +386,37 @@ KRATOS_TEST_CASE_IN_SUITE(ExpectFinalizeSolutionStepCalledOnceForEveryStep, Krat
                      solver_strategy->GetCountFinalizeSolutionStepCalled());
 }
 
+KRATOS_TEST_CASE_IN_SUITE(ExpectOutputIsInitializedAndFinalizedWhenRunCompletesOk, KratosGeoMechanicsFastSuite)
+{
+    TimeLoopExecutor executor;
+    const auto       wanted_num_of_cycles_per_step = std::size_t{1};
+    executor.SetTimeIncrementor(std::make_unique<FixedCyclesTimeIncrementor>(wanted_num_of_cycles_per_step));
+    auto solver_strategy = std::make_shared<DummySolverStrategy>(TimeStepEndState::ConvergenceState::converged);
+    executor.SetSolverStrategyWrapper(solver_strategy);
+    executor.Run(TimeStepEndState{});
+    KRATOS_EXPECT_EQ(solver_strategy->GetCountInitializeOutputCalled(), 1);
+    KRATOS_EXPECT_EQ(solver_strategy->GetCountFinalizeOutputCalled(), 1);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ExpectOutputIsInitializedAndFinalizedWhenRunThrows, KratosGeoMechanicsFastSuite)
+{
+    TimeLoopExecutor executor;
+    const auto       wanted_num_of_cycles_per_step = std::size_t{1};
+    executor.SetTimeIncrementor(std::make_unique<FixedCyclesTimeIncrementor>(wanted_num_of_cycles_per_step));
+    auto solver_strategy = std::make_shared<DummySolverStrategy>(TimeStepEndState::ConvergenceState::converged);
+    solver_strategy->SetOutputProcessCallback([](){throw Exception{"Test exception"};});
```

Comment:
Since we always use KRATOS_ERROR in the code, it might be nice to have that here as well.

Reply:
What I like about explicitly writing `throw Exception` is there is no doubt what kind of exception needs to be caught later in the test. In other words, when you would use `KRATOS_ERROR` you would need to know that it throws a `Kratos::Exception`. In production code we use `KRATOS_TRY` and `KRATOS_CATCH` for exception handling, but those macros hide the exception type that is being thrown. I can't use `KRATOS_CATCH` here, since it **rethrows** the exception after having caught it. For clarity, I prefer how it is currently expressed, but if you have convincing arguments to change it, I would like to learn about them.

Reply:
Good point, no follow-up needed, thanks!